src/transaction.ts

```diff
@@ -612,10 +611,14 @@ export class Snapshot extends EventEmitter {
       });
     };
 
-    return partialResultStream(makeRequest, {
+    return partialResultStream(this._wrapWithIdWaiter(makeRequest), {
       json,
       jsonOptions,
       maxResumeRetries,
+    }).on('response', response => {
```

Comment:
What would happen if the call fails?
As per the detailed design [doc](https://docs.google.com/document/d/11A3r8FG5bTmtccVhtO77xlTNKo_kX0MbdiVhGzEoWeQ/edit#heading=h.nxddek4w5dli) we should make an explicit begin transaction call if the call fails.

Reply:
if the call fails, the transaction is created explicitly with "BeginTransaction" rpc call.
It's covered in the test: **"should use beginTransaction on retry"**.

Reply:
Is this done at the backend or do we explicitly call begin transaction from the client side?